x-pack/legacy/plugins/code/public/components/integrations/code_integrator.tsx

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useState } from 'react';
+import { EuiButtonEmpty, EuiPopover, EuiText } from '@elastic/eui';
+
+import { RepoSelector } from './repo_selector';
+
+interface Props {
+  onRepoSelect: (repo: string) => void;
+  onImportSuccess: (repo: string) => void;
+  repos: string[];
+}
+
+export const CodeIntegrator = ({ onRepoSelect, onImportSuccess, repos }: Props) => {
+  const [showSelector, setShowSelector] = useState(false);
+
+  const handleClick = () => setShowSelector(true);
+
+  const handleSelect = (codeId: string) => {
+    onRepoSelect(codeId);
+    setShowSelector(false);
+    // TODO: show success
```

Comment:
are we going to implement the success notification in this PR. do you think it's currently blocked by the case that we haven't decided on if we should issue the REST API inside the component or not?

Reply:
Currently, the import calls are stubbed out, but the components are written under the assumption that they'll be making the calls themselves.

We could certainly allow the consumer to make the call themselves with a simpler callback, but then CodeIntegrator's not doing much besides generating HTML.

We should discuss both approaches with APM and go with whatever they think is best, but I think both will need the simpler component that is callback-agnostic.